packages/providers/mgt-teamsfx-provider/src/TeamsFxProvider.ts

```diff
@@ -143,34 +79,27 @@ export class TeamsFxProvider extends IProvider {
    * @memberof TeamsFxProvider
    */
   public async getAccessToken(): Promise<string> {
-    let accessToken = await this.credential.getToken(this.scopes);
-    return accessToken ? accessToken.token : '';
-  }
-
-  /**
-   * Update scopes
-   *
-   * @param {string[]} scopes
-   * @memberof TeamsFxProvider
-   */
-  public updateScopes(scopes: string[]) {
-    this.scopes = scopes;
+    try {
+      const accessToken = await this.credential.getToken(this.scopes);
+      this._accessToken = accessToken ? accessToken.token : '';
+      if (!this._accessToken) {
+        throw new Error('Access token is null');
+      }
+    } catch (error) {
+      this.setState(ProviderState.SignedOut);
+      this._accessToken = '';
+    }
+    return this._accessToken;
   }
 
   /**
-   * Performs the internal login using TeamsFx
+   * Performs the login using TeamsFx
    *
    * @returns {Promise<void>}
    * @memberof TeamsFxProvider
    */
-  private async internalLogin(): Promise<void> {
-    let token: string = '';
-
-    try {
-      token = await this.getAccessToken();
-    } catch (error) {
-      console.error("Can't get the access token");
-    }
+  public async login(): Promise<void> {
+    const token: string = await this.getAccessToken();
 
     if (!token) {
       await this.credential.login(this.scopes);
```

Comment:
When discussing, you were saying it was better to call the login within the caller, but now it's in the provider. I prefer the approach where we abstract it all for the developer so we perform the login in the provider like here. Are you onboard with this?

Reply:
Yes, just as you mentioned in teams channel, the login function is only used for mgt-login component, I put it here to make sure mgt-login component works fine.

The login code user write would same as before in a function called when click button:

```ts
async loginBtnClick() {
    try {
      await this.credential.login(this.scope);
      Providers.globalProvider.setState(ProviderState.SignedIn);
      this.setState({
        showLoginPage: false
      });
    } catch (err) {
      alert("Login failed: " + err);
      return;
    }

<Button primary onClick={() => this.loginBtnClick()}>Start</Button>

```

Reply:
Does it mean that every time a user opens the app, they will have to login with the button even though they already consented to the underlying scopes? I can understand the first time, but I feel that from a user experience standpoint, they should not have to login every time... Thoughts @nmetulev @zhenyasav?

Reply:
@sebastienlevert ,  user doesn't need to login every time, because teamsfx SDK and backend auth service will handle the login state, and the next time user open the App, it will show everything without login button.

You can also try the sample to better understanding the flow:

1.	Download and install latest Teams Toolkit extension in VSCode: [Teams Toolkit (Preview) - Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=TeamsDevApp.ms-teams-vscode-extension)

2.	Open an empty folder, and click Teams Toolkit Panel -> View Sample
 
![image](https://user-images.githubusercontent.com/5545529/147426146-250d5090-2d0c-4ec8-bc86-be7b0791fb14.png)

3.	In the sample gallery WebView, you can see the graph toolkit sample

![image](https://user-images.githubusercontent.com/5545529/147426195-f50738b0-333b-409d-9670-738ff0d42b38.png)

4.	Click Create button, after scaffolding finish, you can click F5 to view this sample in Teams